frontend/src/components/annotation/annotation_core.vue

```diff
@@ -1098,26 +1167,17 @@ export default Vue.extend({
   },
   watch: {
     instance_list: function(newVal) {
-      if ((this.task && this.task.file && this.task.file.type === "image" )|| (this.file && this.file.type === "image")) {
-        this.instance_store.set_instance_list(this.file_id, newVal)
-        if(this.task){
-          this.instance_store.set_file_type(this.file_id, this.task.file.type)
-        } else{
-          this.instance_store.set_file_type(this.file_id, this.file.type)
-        }
-
+      if (this.working_file.type === "image") {
+        this.instance_store.set_instance_list(this.working_file.id, newVal)
+        this.instance_store.set_file_type(this.working_file.id, this.working_file.type)
       }
     },
     instance_buffer_dict: {
       deep: true,
       handler: function(newVal) {
-        if ((this.task && this.task.file && this.task.file.type === "video") || (this.file &&this.file.type === "video")) {
-          this.instance_store.set_instance_list(this.file_id, newVal)
-          if(this.task){
-            this.instance_store.set_file_type(this.file_id, this.task.file.type)
-          } else{
-            this.instance_store.set_file_type(this.file_id, this.file.type)
-          }
+        if (this.working_file.type === "video") {
```

Comment:
Really cool! The `working_file` variable makes condition a lot more readable

Reply:
Yeah, it's been set on the factory and now annotation_core can just do working_file.type for both task and file